### DISCUSSION TITLE
Import Dexie Database with Quasar/Electron

### DISCUSSION THREAD

**Discussion by CynthiaBlue** [quasarframework/quasar#13865] [CLI - Electron mode]
I need to be able to export and import my Dexie ( indexedDB) datase to/from my Quasar Electron app.  I need this in case my users' application crashes and they need to restore.

Following the Dexie docs, I've been able to export the whole database to a JSON file, which works great.

I'm trying to import it back, but I'm pretty stuck with this step.
The Dexie docs give a codepen example: 
[https://codepen.io/dfahlander/pen/RqwoaB/](https://codepen.io/dfahlander/pen/RqwoaB/)

Which is great, but it doesn't help me with using Quasar. It uses a dropzone, and I see that Quasar does have a q-uploader component, but I'm not sure how to convert the dropzone in the above example to the equivalent of the Quasar q-uploader.

So my questions include, but are not limited to:
Should I use the q-uploader for this purpose?

When I try a basic q-uploader (and I know I'm not using it right), like this:
```
q-uploader
    url="http://localhost:4444/upload"
    label="Restricted to JSON"
    accept=".JSON"
    style="max-width: 300px"
    >
</q-uploader>
```

I get the following errors:
```
[Vue warn]: Component emitted event "uploading" but it is neither declared in the emits option nor as an "onUploading" prop.
[Vue warn]: Component emitted event "failed" but it is neither declared in the emits option nor as an "onFailed" prop.
```

But I don't see, on the Quasar q-uploder component page, much information on the events "uploading" and "failed".  Where do I find more information about these?

Or do I need to even address those since I'm going to import the contents of the JSON file directly into Dexie? It's not like I'm saving a file onto a server or anything.

I'd appreciate any help.

**Comment by metalsadman** [comment]
update your quasar version (`quasar upgrade -i`), just tested it and i'm not having those warnings.
